src/app/clusters/electrical-energy-measurement-server/ElectricalEnergyMeasurementCluster.h

```diff
@@ -17,46 +17,100 @@
-#include <app/AttributeAccessInterface.h>
+#include <app/server-cluster/DefaultServerCluster.h>
+#include <app/server-cluster/OptionalAttributeSet.h>
+#include <clusters/ElectricalEnergyMeasurement/AttributeIds.h>
+#include <clusters/ElectricalEnergyMeasurement/ClusterId.h>
+#include <clusters/ElectricalEnergyMeasurement/Structs.h>
 #include <lib/core/Optional.h>
 
 namespace chip {
 namespace app {
 namespace Clusters {
 namespace ElectricalEnergyMeasurement {
 
-enum class OptionalAttributes : uint32_t
+// Data structure to hold measurement data for backwards compatibility
+struct MeasurementData
 {
-    kOptionalAttributeCumulativeEnergyReset = 0x1,
+    Structs::MeasurementAccuracyStruct::Type measurementAccuracy;
+    Optional<Structs::EnergyMeasurementStruct::Type> cumulativeImported;
+    Optional<Structs::EnergyMeasurementStruct::Type> cumulativeExported;
+    Optional<Structs::EnergyMeasurementStruct::Type> periodicImported;
+    Optional<Structs::EnergyMeasurementStruct::Type> periodicExported;
+    Optional<Structs::CumulativeEnergyResetStruct::Type> cumulativeReset;
 };
+}; // namespace ElectricalEnergyMeasurement
 
-class ElectricalEnergyMeasurementAttrAccess : public AttributeAccessInterface
+class ElectricalEnergyMeasurementCluster : public DefaultServerCluster
 {
 public:
-    ElectricalEnergyMeasurementAttrAccess(BitMask<Feature> aFeature, BitMask<OptionalAttributes> aOptionalAttrs) :
-        app::AttributeAccessInterface(Optional<EndpointId>::Missing(), app::Clusters::ElectricalEnergyMeasurement::Id),
-        mFeature(aFeature), mOptionalAttrs(aOptionalAttrs)
-    {}
+    // Type aliases for shorter struct type names
+    using MeasurementAccuracyStruct   = ElectricalEnergyMeasurement::Structs::MeasurementAccuracyStruct::Type;
+    using EnergyMeasurementStruct     = ElectricalEnergyMeasurement::Structs::EnergyMeasurementStruct::Type;
+    using CumulativeEnergyResetStruct = ElectricalEnergyMeasurement::Structs::CumulativeEnergyResetStruct::Type;
 
-    ~ElectricalEnergyMeasurementAttrAccess() { Shutdown(); }
+    using OptionalAttributesSet = OptionalAttributeSet<                        //
+        ElectricalEnergyMeasurement::Attributes::CumulativeEnergyImported::Id, //
+        ElectricalEnergyMeasurement::Attributes::CumulativeEnergyExported::Id, //
+        ElectricalEnergyMeasurement::Attributes::PeriodicEnergyImported::Id,   //
+        ElectricalEnergyMeasurement::Attributes::PeriodicEnergyExported::Id,   //
+        ElectricalEnergyMeasurement::Attributes::CumulativeEnergyReset::Id     //
+        >;
 
-    CHIP_ERROR Init();
-    void Shutdown();
+    struct Config
+    {
+        constexpr Config(EndpointId endpoint, BitMask<ElectricalEnergyMeasurement::Feature> featureFlags,
+                         OptionalAttributesSet & optionalAttributes) :
+            endpointId(endpoint),
+            mFeatureFlags(featureFlags), mEnabledOptionalAttributes(optionalAttributes)
```

Comment:
Here we should be computing `mEnabledOptionalAttributes` from `featureFlags` and a one-bit thing.